api/v1/server/handlers/files/create.go

```diff
@@ -0,0 +1,55 @@
+package files
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/labstack/echo/v4"
+
+	"github.com/hatchet-dev/hatchet/api/v1/server/oas/gen"
+	"github.com/hatchet-dev/hatchet/api/v1/server/oas/transformers"
+	"github.com/hatchet-dev/hatchet/pkg/repository"
+	"github.com/hatchet-dev/hatchet/pkg/repository/prisma/db"
+)
+
+func (t *FileService) FileCreate(ctx echo.Context, request gen.FileCreateRequestObject) (gen.FileCreateResponseObject, error) {
+	tenant := ctx.Get("tenant").(*db.TenantModel)
+
+	if !t.config.BlobStorage.Enabled() {
+		return nil, fmt.Errorf("blob storage is not enabled")
+	}
+
+	// marshal the data object to bytes
+	dataBytes, err := json.Marshal(request.Body.Data)
+	if err != nil {
+		return nil, err
+	}
+
+	context := ctx.Request().Context()
+	key := *request.Body.Filename
```

Comment:
do we trust filenames?
should we tenant scope bucket paths?
how do we handle collisions?

Reply:
agreed. we shouldn't be trusting them and should be:
- tenant scoping bucket paths
- setting fileName as a random UUID
- date scoping files to a folder to a certain time frame, probably days.